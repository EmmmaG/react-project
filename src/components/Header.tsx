import * as React from 'react'
import { withStyles, WithStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import space from '../assets/space-photo.jpg'
import down from '../assets/down-chevron.svg'

const styles = () => ({
	headerContainer: {
		border: '1px solid',
		backgroundImage: `url(${space})`,
		backgroundSize: '100% 125%',
		height: '410px',
		position: 'relative' as 'relative',
		width: '100%',
	},
	headerTitle: {
		color: 'white',
		fontSize: '14px',
		fontFamily: 'Lato, sans-serif',
		position: 'absolute' as 'absolute',
		top: '18px',
		left: '40px',
	},
	headerMainText: {
		color: 'white',
		fontSize: '50px',
		fontFamily: 'Lato, sans-serif',
		marginTop: '135px',
	},
	arrow: {
		height: '20px',
		bottom: '30px',
		position: 'absolute' as 'absolute',
	},
})

export interface Props extends WithStyles<typeof styles> {}

export class Header extends React.Component<Props> {
	public render() {
		const { classes } = this.props

		return (
			<div className={classes.headerContainer}>
				<Typography className={classes.headerTitle}>SPACE SAVVY</Typography>
				<Typography className={classes.headerMainText}>Discover Space Missions</Typography>
				<img src={down} className={classes.arrow} />
			</div>
		)
	}
}

export default withStyles(styles)(Header)
