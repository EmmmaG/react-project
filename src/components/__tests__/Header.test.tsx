import * as React from 'react'
import { shallow } from 'enzyme'

import Header, { Props } from '../Header'

const props: Props = {
	classes: {} as Record<string, string>,
}

describe('Header', () => {
	it('renders correctly', () => {
		const wrapper = shallow(<Header {...props} />)
		expect(wrapper).toMatchSnapshot()
	})
})
