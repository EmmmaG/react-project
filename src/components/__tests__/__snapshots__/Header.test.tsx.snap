// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<Header
  classes={
    Object {
      "arrow": "Header-arrow-5",
      "arrowContainer": "Header-arrowContainer-4",
      "headerContainer": "Header-headerContainer-1",
      "headerMainText": "Header-headerMainText-3",
      "headerTitle": "Header-headerTitle-2",
    }
  }
/>
`;
